### ISSUE TITLE
outdent lib-module check for 'module' breaks code bundled with webpack

### ISSUE THREAD

**Issue by avaragado** [cspotcode/outdent#9]
I'm not _entirely_ sure what's going on, but I'm using `outdent` in a project that's built using [next.js](https://github.com/zeit/next.js) (and hence webpack), and the built code has broken on upgrading from outdent 0.3.0 to 0.5.0.

The browser error includes these lines:

```
ReferenceError: exports is not defined
    at Object.<anonymous> (http://localhost:3000/_next/-/page/foo.js:7394:30)
    at Object../node_modules/outdent/lib-module/index.js (http://localhost:3000/_next/-/page/foo.js:7401:30)
    at __webpack_require__ (http://localhost:3000/_next/-/manifest.js:715:31)
```
The JS built by webpack incorporates the `outdent/lib-module/index.js` file like this (I've omitted the irrelevant bits of the `foo.js` file):

```js
/***/ "./node_modules/outdent/lib-module/index.js":
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
/* WEBPACK VAR INJECTION */(function(module) {/* unused harmony export outdent */
// In the absence of a WeakSet or WeakMap implementation, don't break, but don't cache either.

// ...lots omitted here...

// Named exports.  Simple and preferred.
/* harmony default export */ __webpack_exports__["a"] = (outdent);

if (true) {
    module.exports = exports = outdent;
    // TODO is this necessary?
    Object.defineProperty(outdent, '__esModule', { value: true });
    outdent.default = outdent;
    outdent.outdent = outdent;
}
//# sourceMappingURL=index.js.map
/* WEBPACK VAR INJECTION */}.call(__webpack_exports__, __webpack_require__("./node_modules/webpack/buildin/harmony-module.js")(module)))

```

You'll see that the original code's check `if (typeof module !== 'undefined')` has been optimised to `if (true)`, presumably as `module` exists -- but there's no `exports` variable, so the code explodes.

Is there a better check you could do on that line to achieve the desired effect?

**Comment by cspotcode** [comment]
Thanks for reporting this.

Shouldn't webpack be declaring an `exports` object?  It's common for Javascript code to look like this:

```
exports.foo = function() { /* do stuff */ }
```

It looks like webpack is rewriting all instances of `exports` into `__webpack_exports__`, but it's missing one.  Do you know why that's happening?  Can the webpack folks confirm or deny this behavior?

**Comment by cspotcode** [comment]
@avaragado I think this'll work:

https://github.com/cspotcode/outdent/blob/master/src/index.ts#L179

Change the line to:

    if(typeof module !== 'undefined' && typeof exports !== 'undefined') {

Hopefully it'll trigger webpack to inject both the module and exports variables.

Alternatively, it might be possible to remove the `exports` assignment.  The important thing is replacing `module.exports`.

I don't have time to investigate, but if you feel like doing a quick testing and sending a pull request, I'll be sure to publish a new version to npm.

**Comment by chasemccoy** [comment]
Any updates on fixing this @cspotcode?

**Comment by cspotcode** [comment]
@chasemccoy I'm swamped at work, but if someone sends me a tested PR, I'll merge and publish.

**Comment by chasemccoy** [comment]
Opened #13 to fix this.

**Comment by cspotcode** [comment]
Should be fixed in v0.6.0.  Thanks again for reporting, and let me know if you encounter this error again with webpack or another bundler.

**Comment by chasemccoy** [comment]
Thanks so much!
